.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/ChineseDateParserConfiguration.cs

```diff
@@ -739,6 +739,22 @@ private int ConvertChineseYearToInteger(string yearChsStr)
             var year = 0;
             var num = 0;
 
+            var regionTitleMatch = this.config.RegionTitleRegex.MatchExact(yearChsStr, trim: true);
+            if (regionTitleMatch != null && regionTitleMatch.Value != string.Empty)
+            {
+                // handle "康熙元年" refer to https://zh.wikipedia.org/wiki/%E5%B9%B4%E5%8F%B7
+                int basicYear = this.config.DynastyYearMap[regionTitleMatch.Value];
+                string biasYearStr = yearChsStr.Substring(regionTitleMatch.Value.Length, yearChsStr.Length - regionTitleMatch.Length);

```

Comment:
Instead of parsing the value like these, can't named groups be used in the regex so you capture what you need?